runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -224,6 +225,9 @@
    */
   public static final String PROJECT_ID_REGEXP = "[a-z][-a-z0-9:.]+[a-z0-9]";
 
+  /** Dataflow service endpoints are expected to match this pattern. */
+  static final String ENDPOINT_REGEXP = "https://[\\S]*googleapis\\.com[/]?";
```

Comment:
Is this always true? (I do not know the answer but probably good to check wrt staging etc. endpoints.)

Reply:
I copied this from the Python SDK, so I'm not sure what the source of truth is. But a quick code search doesn't return any obvious negatives.

https://github.com/apache/beam/blob/ac003b6250f8b6e7dd8455fd5ffea7ded3b85bcd/sdks/python/apache_beam/options/pipeline_options_validator.py#L106